frontend/lib/src/FileUploadClient.ts

```diff
@@ -101,19 +111,38 @@ export class FileUploadClient {
     widget: WidgetInfo,
     fileUploadUrl: string,
     file: File,
-    onUploadProgress?: (progressEvent: any) => void,
-    cancelToken?: CancelToken
+    _onUploadProgress?: (progressEvent: any) => void,
+    _cancelToken?: CancelToken
   ): Promise<void> {
     this.offsetPendingRequestCount(widget.formId, 1)
-    return this.endpoints
-      .uploadFileUploaderFile(
-        fileUploadUrl,
-        file,
-        this.sessionInfo.current.sessionId,
-        onUploadProgress,
-        cancelToken
-      )
-      .finally(() => this.offsetPendingRequestCount(widget.formId, -1))
+
+    // stlite Modification: Use form upload
+    const form = new FormData()
+    form.append("sessionId", this.sessionInfo.current.sessionId)
```

Comment:
Can this be deleted?
Looks like it's not referred to both in Streamlit and stlite.
Streamlit: https://github.com/streamlit/streamlit/blob/1.27.0/lib/streamlit/web/server/upload_file_request_handler.py#L84

Reply:
-> Suggested the fix in https://github.com/whitphx/streamlit/pull/8.

I think the upstream code can also be fixed here, https://github.com/streamlit/streamlit/blob/c7d29c1ade36a347fd740e4adb66e06708957159/frontend/app/src/connection/DefaultStreamlitEndpoints.ts#L120
because the server-side handler doesn't look at this field as https://github.com/streamlit/streamlit/blob/develop/lib/streamlit/web/server/upload_file_request_handler.py#L84

Reply:
Yes, indeed. This can be cleaned up 👍

Reply:
Created a PR -> https://github.com/streamlit/streamlit/pull/7980